src/main/scala/emm/effects/Binder.scala

```diff
@@ -29,6 +30,22 @@ trait BinderLowPriorityImplicits {
       NAP.pack(back)
     }
   }
+
+  implicit def binderState[S, C <: Effects](implicit C: Binder[C], T: Traverser[C], NN: NonNested[C]): Binder[State[S, ?] |: C] = new Binder[State[S, ?] |: C] {
```

Comment:
Another realization. `StateT` is *almost* just wrapping `Kleisli` in the same `F[_]` the `Kleisli` returns. The difference being, the tuple in the return type. But that's just a specific case that we might not need to match on individually. Meaning, perhaps we could match `StateT` on something like `F |: Kleisli |: Base`.